src/all/mangapluscreators/src/eu/kanade/tachiyomi/extension/all/mangapluscreators/MangaPlusCreators.kt

```diff
@@ -16,94 +18,200 @@ import okhttp3.Request
 import okhttp3.Response
 import rx.Observable
 import uy.kohesive.injekt.injectLazy
+import java.text.SimpleDateFormat
+import java.util.Locale
 
 class MangaPlusCreators(override val lang: String) : HttpSource() {
 
     override val name = "MANGA Plus Creators by SHUEISHA"
 
-    override val baseUrl = "https://medibang.com/mpc"
+    override val baseUrl = BASE_URL
 
     override val supportsLatest = true
 
     override fun headersBuilder(): Headers.Builder = Headers.Builder()
-        .add("Origin", baseUrl.substringBeforeLast("/"))
         .add("Referer", baseUrl)
         .add("User-Agent", USER_AGENT)
 
     private val json: Json by injectLazy()
 
+    // POPULAR Section
     override fun popularMangaRequest(page: Int): Request {
         val newHeaders = headersBuilder()
             .set("Referer", "$baseUrl/titles/popular/?p=m")
-            .add("X-Requested-With", "XMLHttpRequest")
             .build()
 
-        val apiUrl = "$API_URL/titles/popular/list".toHttpUrl().newBuilder()
-            .addQueryParameter("page", page.toString())
-            .addQueryParameter("pageSize", POPULAR_PAGE_SIZE)
-            .addQueryParameter("l", lang)
-            .addQueryParameter("p", "m")
-            .addQueryParameter("isWebview", "false")
-            .addQueryParameter("_", System.currentTimeMillis().toString())
-            .toString()
+        val popularUrl = "$baseUrl/titles/popular/?p=m&l=$lang".toHttpUrl().toString()
 
-        return GET(apiUrl, newHeaders)
+        return GET(popularUrl, newHeaders)
     }
 
-    override fun popularMangaParse(response: Response): MangasPage {
-        val result = response.asMpcResponse()
+    override fun popularMangaParse(response: Response): MangasPage = parseMangasPageFromElement(
+        response,
+        "div.item-recent",
+    )
 
-        checkNotNull(result.titles) { EMPTY_RESPONSE_ERROR }
+    fun parseMangasPageFromElement(response: Response, selector: String): MangasPage {
+        val result = response.asJsoup()
 
-        val titles = result.titles.titleList.orEmpty().map(MpcTitle::toSManga)
+        val mangas = result.select(selector).map { element ->
+            popularElementToSManga(element)
+        }
 
-        return MangasPage(titles, result.titles.pagination?.hasNextPage ?: false)
+        return MangasPage(mangas, false)
     }
 
+    fun popularElementToSManga(element: org.jsoup.nodes.Element): SManga {
+        val titleThumbnailUrl = element.select(".image-area img").attr("src")
+        val titleContentId = titleThumbnailUrl.toHttpUrl().pathSegments[2]
+        return SManga.create().apply {
+            title = element.select(".title-area .title").text().toString()
```

Comment:
use `selectFirst`

Reply:
aha! yea I did ponder for a bit about how `select` is supposed to be returning a list but it compiled and ran fine so left it as is